src/client/Health/NodeView.jsx

```diff
@@ -153,49 +430,59 @@ const NodeView = ({ status, onRestart, onShutdown }) => {
                 </Button>
               )}
             </Modal>
+
+            <Button
+              id="addsouth"
+              className="inline-button autosize oi-south-button"
+              color={engine?.safeMode ? 'warning' : 'success'}
+              size="sm"
+              outline
+            >
+              + South
+            </Button>
+            <UncontrolledPopover className="pop-container" trigger="legacy" placement="top" target="addsouth">
+              <PopoverHeader>Add a new South connector</PopoverHeader>
+              <PopoverBody>
+                <NewDataSourceRow protocolList={protocolList} addDataSource={addDataSource} />
+              </PopoverBody>
+            </UncontrolledPopover>
+
+            <Button
+              id="addnorth"
+              className="inline-button autosize oi-north-button"
+              color={engine?.safeMode ? 'warning' : 'success'}
+              size="sm"
+              outline
+            >
+              + North
+            </Button>
+            <UncontrolledPopover className="pop-container" trigger="legacy" placement="bottom" target="addnorth">
+              <PopoverHeader>Add a new North connector</PopoverHeader>
+              <PopoverBody>
+                <NewApplicationRow apiList={apiList} addApplication={addApplication} />
+              </PopoverBody>
+            </UncontrolledPopover>
```

Comment:
When the page first loads, apiList is undefined but is required in NewApplicationRow. You must check if apiList is defined before rendering this component. Something like :

{{ apiList ? <UncontrolledPopover className="pop-container" trigger="legacy" placement="bottom" target="addnorth">
              <PopoverHeader>Add a new North connector</PopoverHeader>
              <PopoverBody>
                <NewApplicationRow apiList={apiList} addApplication={addApplication} />
              </PopoverBody>
            </UncontrolledPopover> : null }}

Reply:
You can do the same things for south

Reply:
please include the button too